tests/test_recompile_ux.py

```diff
@@ -0,0 +1,92 @@
+from abc import ABC
+from abc import abstractmethod
+from typing import Any
+from typing import List
+
+import torch
+
+import torchdynamo
+import torchdynamo.config
+import torchdynamo.testing
+
+torchdynamo.config.debug = False
+from torchdynamo.utils import counters
+
+class RecompileUxTests(torchdynamo.testing.TestCase):
+    # TODO(whc) dynamo actualy recompiles one more time than the cache limit
+    max_recompiles = 2
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        torchdynamo.config.cache_size_limit = max(1, RecompileUxTests.max_recompiles - 1)
```

Comment:
This will leak state into other tests.  Perhaps:
```suggestion
        cls._exit_stack.enter_context(
           unittest.mock.patch.object(torchdynamo.config, "cache_size_limit", max(1, RecompileUxTests.max_recompiles - 1))
```

Which is using this:
https://github.com/jansel/torchdynamo/blob/5e575216aa7467cda972cfa4cd3443175ca1d6b6/torchdynamo/testing.py#L159

Reply:
cool, yea that's a handy trick - i'll use it for now.

i'm not sure if I even want to set the cache limit in the test, or if i just want to change the cache limit on master and have the tests verify the intended behavior- so that's another thing I wanted to discuss.

probably makes sense to decide after going a little further with what UX the tests are converging towards